personalities/sctfe/modules/dedup/dedup.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2024 The Tessera authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package dedup limits the number of duplicate entries a personality allows in a Tessera log.
+package dedup
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/transparency-dev/trillian-tessera/api/layout"
+	"github.com/transparency-dev/trillian-tessera/client"
+	"k8s.io/klog/v2"
+)
+
+// KV holds a LeafID and an Idx for deduplication
+type KV struct {
+	K []byte
+	V uint64
+}
+
+type DedupStorage interface {
+	Add(ctx context.Context, kvs []KV) error
+	Get(ctx context.Context, key []byte) (uint64, bool, error)
+}
+
+type LocalDedupStorage interface {
+	Add(ctx context.Context, kvs []KV) error
+	Get(ctx context.Context, key []byte) (uint64, bool, error)
+	LogSize() (uint64, error)
+}
+
+type LocalBEDedup struct {
```

Comment:
If there is no other non-best-effort deduplication, we can name this as `LocalDedup`. The "best effort" can be mentioned in the doc comment.